### ISSUE TITLE
Allow the promotion of entities from one environment to another

### ISSUE THREAD

**Issue by GeorgiChochov** [AbsaOSS/enceladus#594]
## Background
Currently there is no easy way to move entity definitions across environments. Recreating datasets by hand can be especially tedious.

## Feature
It would be great if we could push a button in Menas and promote datasets and their dependencies in bulk to another Menas server, (e.g, Test -> Dev -> UAT -> Pre-prod -> Prod). 
An open question is how to handle conflicts (cases where a dataset in the remote environment already has a dataset with the same name and version). This will conflict resolution should be drawn up in a shared doc and once resolved posted here.

**Comment by Zejnilovic** [comment]
Could the conflicts be solved by the same logic used in the continuous migration issue #447 and PR #778?

**Comment by yruslan** [comment]
Yes and no. Yes, many ideas are applicable, like creating new versions of entities. But there are differences as well:
* Continuous migration relies on ObjectId to be migrated across DB versions. This is not the case for transferring documents between environments.
* Entity mapping helps resolve foreign keys, and since we know the previous version of the database contains almost the same data the entity name-version mapping is stable and can be reused. This is not the case with dataset promotion since each db is completely independent. The design for entity mapping, if used, should be adapted to that.

But transferring a dataset between envs may utilize the export/import feature. After all, transferring a dataset is just export it on one env and import it on another one. A design doc for export/import is pending.

**Comment by benedeki** [comment]
I support the solution via import/export 👍
